File: solutions/clojure/cars-assemble/src/cars_assemble.clj
(ns cars-assemble)

(def hourly-production 221)

(defn success-rate 
  "Returns the success rate given production speed"
  [speed]
  (condp >= speed
        0 0.0
        4 1.0
        8 0.9
        9 0.8
        0.77))

(defn production-rate
  "Returns the assembly line's production rate per hour,
   taking into account its success rate"
  [speed]
  (* speed hourly-production (success-rate speed)))

(defn working-items
  "Calculates how many working cars are produced per minute"
  [speed]
  (int (/ (production-rate speed) 60)))

File: solutions/java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {
    public double productionRatePerHour(int speed) {
        double production = switch (speed) {
            case 5:
            case 6:
            case 7:
            case 8:
                yield 0.9;
            case 9:
                yield 0.8;
            case 10:
                yield 0.77;
            default:
                yield 1.0;
        };
        final int CARS_EACH_HOUR = 221;
        return  speed*CARS_EACH_HOUR*production;
    }

    public int workingItemsPerMinute(int speed) {
        return (int)productionRatePerHour(speed)/60;
    }
} 
 